### PR TITLE
implement autocompleter for custom actions

### PR DESCRIPTION
**Pull request by ulferts**
* Based on jquery ui autocompleter
* wrapps standard select box
* different style for multi/singleselect
* simplifies styles for selected values (attachments, watchers)

Part of (but does not close) https://community.openproject.com/projects/openproject/work_packages/27118

### CODE REVIEW

File: frontend/app/components/common/autocomplete-select-decoration/autocomplete-select-decoration.component.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        [linkTitle]="removeItemText(item)"
+        linkClass="form--selected-value--remover">
+      <op-icon icon-classes="icon-remove"
+               [icon-title]="removeItemText(item)"></op-icon>
+    </accessible-by-keyboard>
+  </li>
+</ul>
+
+<input
+    [ngClass]="{ '-hidden': !isMulti && selectedItems.length }"
```
Comment (oliverguenther) [inline frontend/app/components/common/autocomplete-select-decoration/autocomplete-select-decoration.component.html]: Why not use `[hidden]="..."` ?. We already have styles for that.

File: frontend/app/components/common/autocomplete-select-decoration/autocomplete-select-decoration.component.html [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    type="text"
+    class="form--input -autocomplete"
+    placeholder="{{placeholderText}}">
```
Comment (oliverguenther) [inline frontend/app/components/common/autocomplete-select-decoration/autocomplete-select-decoration.component.html]: Prefer `[attr.placeholder]="placeholderText"` over the interpolated value unless you need to coerce a value to a string. The interpolation style will bind the attribute before its initialized. (For a placeholder, that effect is negligible but I suggest we avoid interpolations unless necessary.)

**Comment by oliverguenther** [comment]
Nice 👍 I would prefer a more compact style ( a tag-like look as in select2 but above the input) since selecting multiple value will otherwise take up a lot of space and there is too much space around each item that makes them appear isolated. 

bildschirmfoto 2018-03-01 um 07 33 46

vs (badly shopped tags from column dropdown): 

tag-look

We'll also have to look into accessibility and add a sequence of `aria-activedescendant` to each result to make them discoverable through arrow keys. There's a popup example here: https://www.w3.org/TR/2017/NOTE-wai-aria-practices-1.1-20171214/examples/combobox/aria1.1pattern/listbox-combo.html

This all does not affect the PR however, we can rework this later on. Merging after the two minor comments are addressed.

**Comment by ulferts** [comment]
The two minor points are addressed. 

Thanks for already offering a proposal to improve accessibility. Before addressing it, let's merge it and get the input of more people. There will probably be a couple of changes to this component before it is done.

I agree with your point that the current implementation takes up a lot of space, and for me at least, it would be ok to change to your more compact style. But if we do that, we should also apply that rule to the other places in the application where we have a similar functionality: watchers and attachments.

**Comment by oliverguenther** [comment]
Fine with me!

**Pull request merged**